### DISCUSSION TITLE
Stack.java içeriği nedir?

### DISCUSSION THREAD

**Discussion by mesutckr** [gusanmaz/BMB212-Algorithms-2223#20] [Q&A] [answered]
6. soru için Stack.java ve CSVReader.java kullanın demişsiniz. Stack.java dosyası repo da yok. bu dosya  ne için kullanılacak ve kod içeriği nedir? bu dosyayı kullanmadan  CSVReader.java kullanılarak soru çözülebiliyor Stack.java ya gerek var mı?

**Comment by gusanmaz** [comment, marked as answer]
`Stack.java`'yi siz yazacaksiniz.

**Reply by gusanmaz** [reply]
Konuyla ilgili 2. soru:

> Stack.java dosyasını oluşturup kendimiz yazmamız gerekiyor. Stack yapısını kullanmamızı istemişsiniz. Bu dosyada javanın Stack sınıfını kullanabilir miyiz. Stack işlemleri yine bu dosya içerisinde olacak..
> import java.util.Stack; 
> 
> Kendimiz yazsak ArrayList Kullanmak mantıklı onu da kullanmayın demişsiniz. diğer türlü string dizi kullanmamız gerekli onda da boyut belirtmek gerek.. hangisi ile bu dosyayı kodlayayım ?

Cevap:

`pop`, `push` metodlari olan Stack verilerini dizide saklayan klasik, basit bir Stack sinifi tanimlamaniz isteniyor.

Odev aciklamalarindaki su uyariyi dikkate alarak diziniz icin uygun bir boyut belirleyebilirsiniz.

> Programiniz ayrica maksimum 1000 satirlik veri iceren CSV dosyalari icin (ornegin [data/path/unix_paths_1000.csv](https://github.com/gusanmaz/BMB212-Algorithms-22-23/blob/main/data/path/unix_paths_1000.csv)) bile dogru cikti uretmelidir.
